### PR TITLE
test: Avoid unnecessary restarts of unmanaged pods

### PR DESCRIPTION
**Pull request by pchaigno**
In tests, after Cilium installations, we often restart unmanaged pods to ensure they are managed by Cilium. In particular, on GKE, we used to restart unmanaged pods from both the `kube-system` and the `cilium` namespaces.

However, commit 48be458 ("test: GKE: Install Cilium in kube-system namespace") removed the cilium namespace to install Cilium in the `kube-system` namespace. One of the two calls to `RestartUnmanagedPodsInNamespace()` is therefore unnecessary. In addition, we also already restart pods in the namespace of the log-gatherer pods for all CI environments (vs. just GKE). If that last namespace is the `kube-system` namespace, then we don't need any call to `RestartUnmanagedPodsInNamespace()` for GKE.

I expect this will fix flake #14915. In that flake, some pods are not found while attempting to restart unmanaged pods. The flake started appearing in master when we merged commit 48be458. The theory is that the two quick calls to `RestartUnmanagedPodsInNamespace()` for the same namespace lead us, in the second call, to select pods that have already
been restarted by the first call. Such pods may disappear between the time we select them and the time we actually execute `kubectl delete`, resulting in the error:

    Error from server (NotFound): pods "kube-dns-66d6b7c877-dp4q2" not found

Fixes: https://github.com/cilium/cilium/pull/14899

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

**Review by aditighag — APPROVED** [review]
The fix looks reasonable to me. But it's good to get a review from ci-structure team.

**Comment by aditighag** [comment]
test-gke
Perhaps run test-gke couple of more times to confirm that it fixes the flake?

**Comment by pchaigno** [comment]
> Perhaps run test-gke couple of more times to confirm that it fixes the flake?

I thought about it, but I think the change makes sense to have regardless of whether it fixes the flake. So we might as well merge quickly rather than validate it does fix the flake. We'll soon know anyway.

**Comment by Skymirrh** [comment]
> I thought about it, but I think the change makes sense to have regardless of whether it fixes the flake.

I agree.

**Review by nbusseneau — APPROVED** [review]

**Comment by aditighag** [comment]
> > Perhaps run test-gke couple of more times to confirm that it fixes the flake?
> 
> I thought about it, but I think the change makes sense to have regardless of whether it fixes the flake. So we might as well merge quickly rather than validate it does fix the flake. We'll soon know anyway.

Yup, makes sense. Hit this on my PR. If the re-run hits the flake again, I'll have to rebase to master once your fix is merged.

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
test-gke

**Comment by pchaigno** [comment]
test-4.19

**Comment by pchaigno** [comment]
test-4.19

**Comment by pchaigno** [comment]
GKE failed again with https://github.com/cilium/cilium/issues/14915's `Error from server (NotFound): pods "kube-dns-66d6b7c877-zttfq" not found` (cf. https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/4311/testReport/junit/Suite-k8s-1/16/K8sKafkaPolicyTest_Kafka_Policy_Tests_KafkaPolicies/). So the current fix isn't enough. I think we should still merge once 4.19 is green. I removed the `Fixes:` tag in the original post to not close https://github.com/cilium/cilium/issues/14915.

**Comment by pchaigno** [comment]
This is ready to merge. Failure is known flake https://github.com/cilium/cilium/issues/14915, reviews are in.

**Pull request merged**
